### PR TITLE
Only include node states that are meterable in metering

### PR DESCRIPTION
**Pull request by oyving**
Maintain a list of node states that are meterable, and make sure we include only these in our metering snapshots. Unfortunately, we maintain this list inside the Metering maintainer and not on the node state class itself. The reason for this is that we have _many_ different `Node.State` and `NodeState` classes that do exactly the same.

### CODE REVIEW

File: controller-server/src/main/java/com/yahoo/vespa/hosted/controller/maintenance/ResourceMeterMaintainer.java [hunk 1]
```diff
@@ -85,9 +87,23 @@ protected void maintain() {
...
+    }
+
+    /**
+     * Checks if the node is in some state where it is in active use by the tenant,
+     * and not transitioning out of use, in a failed state, etc.
+     */
+    private static final Set<Node.State> METERABLE_NODE_STATES = Set.of(
+            Node.State.reserved,   // an application will soon use this node
+            Node.State.active,     // an application is currently using this node
+            Node.State.inactive    // an application owner might set a node inactive
```
Comment (olaaun) [inline controller-server/src/main/java/com/yahoo/vespa/hosted/controller/maintenance/ResourceMeterMaintainer.java]: Nitpick, but I believe application owner has no control over deactivation of nodes?

**Review by olaaun — APPROVED** [review]
LGTM👍

File: controller-server/src/main/java/com/yahoo/vespa/hosted/controller/maintenance/ResourceMeterMaintainer.java (hunk 1 above)
Comment (freva) [reply controller-server/src/main/java/com/yahoo/vespa/hosted/controller/maintenance/ResourceMeterMaintainer.java]: Correct, moved to inactive by configserver when the node is no longer needed (retired manually, requested flavor changed, number of nodes changed).

Comment (oyving) [reply controller-server/src/main/java/com/yahoo/vespa/hosted/controller/maintenance/ResourceMeterMaintainer.java]: Aha! But they are assigned to a user, and just waiting for clean-up of the node?

Comment (freva) [reply controller-server/src/main/java/com/yahoo/vespa/hosted/controller/maintenance/ResourceMeterMaintainer.java]: Yes, they are assigned to the user and just waiting to be expired, the purpose of that state is if the user deploy unintended change and starts losing nodes, they can undo it without losing data (the inactive nodes will be assigned back to them)

**Comment by oyving** [comment]
@olaaun and @freva - I updated the comment based on your feedback. Don't want any misconceptions lingering in the code. Looks good now?

**Review by freva — APPROVED** [review]

**Review by olaaun — APPROVED** [review]

**Pull request merged**